### ISSUE TITLE
Car build list

### ISSUE THREAD

**Issue by Scholesy46** [TJSteel/OnlineRacingManager#90]
When a car is added to a series can we have the ability to go in and give the car a build list listing PI, weight, power and parts required

**Comment by TJSteel** [comment]
Yeah, I was thinking of adding this in at some point but wasn't quite sure how to do it because of the huge variation in the types of parts you can apply to a car

One thing I was thinking of was just having the ability to add / remove an upgrade, where you select the upgrade group, and a text field where you would just type the part details,
if you take the spec miata build for example: http://forums.theonlineracingassociation.com/t20284-spec-miata-build-rules

you could have the following 2 for the first item
(Dropdown list) (Category) - Engine
(Text box) (Description) - Air Filter - Race

Or, if you could get a list of every different type of part that would be used (could always add to the list later if we miss something) we could do it like this
(Dropdown list) (Category) - Engine
(Dropdown list) (Part) - Air Filter
(Dropdown list) (Type) - Race

Or maybe easier would be grouping the Part and Type like this
(Dropdown list) (Category) - Engine
(Dropdown list) (Part) - Air Filter - Race

what do you think @Scholesy46

**Comment by Scholesy46** [comment]
I would go for option 1, Dropdown catagory, text box description as there are hundreds of upgrade options and getting them all would be a nightmare. If the text box could also be edited after being added that would be fantastic due to certain events receiving BoP changes

**Comment by TJSteel** [comment]
What if, I make it work the way Entering a team does? you get a list of available parts, if one hasn't been created yet, you just add a new one, that would then put it in the list to select from?

I'll have a think after lunch if I got time, and see what I can come up with, for this we can't just get a full list of them, so, we want to make it so that we can select them easy without having to type the same shit everytime, but we also want it to by dynamic enough to allow us to add additional options

I don't think a full list would take that long to build up either to be honest, obviously we wouldn't validate if it's actually available on that car, we'd just allow you to put say, remove restrictors on everything, even if you're trying to put it on a Miata.

**Comment by Scholesy46** [comment]
Yea that could work actually. would it be possible when you've added the parts or them to show up like the following:

(Year) (Manufacture) (Model)
PI - 567
Power - 467 bhp
Torque - 359 ft/lb
Weight - 3,546 lb

**Conversion**
Engine Swap - Racing V8
Drivetrain Swap - RWD Conversion

**Engine**
Air Filter - Race
Exhaust - Race
Flywheel - Sport

**Performance and Handing**
Brakes - Race
Roll Cage - Race

**Driveline**
Transmission - Sport
Differential - Race

**Wheels**
Compound - Sport
Front Tyre Width - 180/15R40
Rear Tyre Width - 195/15R35

**Restrictions**
Power - 10%
Rear Grip - 5%

Power displayed is before the 10% restriction is applied and should equal the Power displayed in imperial when viewing the car in your garage

**Comment by TJSteel** [comment]
yeah definitely, I'll load up the game tonight and go through and find all the upgrades I can think of, I'll see if I can get a db design done for this today at least though, I've a good idea of what will be needed anyway

**Comment by TJSteel** [comment]
I think this should be enough to get the build list going in terms of db design

![image](https://user-images.githubusercontent.com/35134286/76088592-2d78dd80-5fb0-11ea-8474-f2920e1036b1.png)

so basically, for the build, you just add a build option to the car from a single dropdown (or I could put 2 dropdowns to filter first by category but I don't think that would be necessary if sorted well), if the option you want to add isn't there, you would just need to add this on a page made for editing the build options table.
